packages/workflow/src/internals.ts

```diff
@@ -611,7 +617,12 @@ export class Activator implements ActivationHandler {
 
   // Intentionally non-async function so this handler doesn't show up in the stack trace
   protected queryWorkflowNextHandler({ queryName, args }: QueryInput): Promise<unknown> {
-    const fn = this.queryHandlers.get(queryName)?.handler;
+    let fn = this.queryHandlers.get(queryName)?.handler;
+    // Fallback to default query handler if no handler is registered.
+    if (fn === undefined) {
+      fn = this.defaultQueryHandler;
```

Comment:
The default query handler will need an extra argument, the first one, that indicates the name of the query being called. See `signalWorkflowNextHandler()`.

Reply:
Fixed, improved the test to catch this case. Thanks